primitives/header-chain/src/lib.rs

```diff
@@ -0,0 +1,35 @@
+// Copyright 2019-2020 Parity Technologies (UK) Ltd.
+// This file is part of Parity Bridges Common.
+
+// Parity Bridges Common is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity Bridges Common is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity Bridges Common.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Defines traits which represent a common interface for Substrate pallets which want to
+//! incorporate bridge functionality.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use frame_support::Parameter;
+
+/// A base trait for pallets which want to keep track of a full set of headers from a bridged chain.
+pub trait MinimalHeaderChain {
+	/// Transaction type.
+	type Transaction: Parameter;
```

Comment:
Are these bounds needed? Can we get away with no `frame-support` dependency?

Reply:
Yeah they are needed. The Substrate macros complain if we don't have them

Reply:
Well it probably compliants about a bunch of traits, but not this specific one :) We can simply import `Codec + EncodeLike + Clone + Eq + core::fmt::Debug` and it should have the same effects without extra `frame_support` dependency, which surprisingly is pretty heavy.

Reply:
Lol, kinda gross but sure I can change that